creator-node/src/routes/tracks.js

```diff
@@ -550,26 +551,101 @@ module.exports = function (app) {
+        let trackRecord = await libs.Track.getTracks(1, 0, [blockchainId])
+        if (!trackRecord || trackRecord.length === 0 || !trackRecord[0].hasOwnProperty('blocknumber')) {
+          // TODO - return a server err response
+          return sendResponse(req, res, errorResponseServerError('Missing or malformatted track fetched from discovery node.'))
+        }
+
+        trackRecord = trackRecord[0]
+
+        // query the files table for the metadata multihash from discovery
+        // no CNodeUserUUID, but because this track is associated with a user and contains that user_id
+        // inside it, it's unique to this user
+        const file = await models.File.findOne({ where: { multihash: trackRecord.metadata_multihash } })
+        if (!file) {
+          return sendResponse(req, res, errorResponseServerError('Missing or malformatted track fetched from discovery node.'))
+        }
+
+        // read the metadata multihash from from the file system
+        let metadataJSON
+
+        const fileBuffer = await readFile(file.storagePath)
+        metadataJSON = JSON.parse(fileBuffer)
+        if (
+          !metadataJSON ||
+          !metadataJSON.track_segments ||
+          !Array.isArray(metadataJSON.track_segments) ||
+          !metadataJSON.track_segments.length
+        ) {
+          return sendResponse(req, res, errorResponseServerError(`Malformatted metadataJSON stored for metadata multihash ${trackRecord.metadata_multihash}.`))
+        }
+
+        // make sure all track segments have the same sourceFile
+        const segments = metadataJSON.track_segments.map(segment => segment.multihash)
+
+        let fileSegmentRecords = await models.File.findAll({
+          attributes: ['sourceFile'],
+          where: {
+            multihash: segments,
+            cnodeUserUUID: file.cnodeUserUUID
+          },
+          raw: true
+        })
+
+        // check that the number of files in the Files table for these segments for this user matches the number of segments from the metadata object
+        if (fileSegmentRecords.length !== metadataJSON.track_segments.length) {
+          return sendResponse(req, res, errorResponseServerError(`Track content mismatch for blockchainId ${blockchainId} - number of segments don't match between local and discovery`))
+        }
+
+        // check that there's a single sourceFile that all File records share by getting an array of uniques
+        const uniqSourceFiles = fileSegmentRecords.map(record => record.sourceFile).filter((v, i, a) => a.indexOf(v) === i)
+
+        if (uniqSourceFiles.length !== 1) {
+          return sendResponse(req, res, errorResponseServerError(`Track content mismatch for blockchainId ${blockchainId} - there's not one sourceFile that matches all segments`))
+        }
+
+        // search for the copy320 record based on the sourceFile
+        fileRecord = await models.File.findOne({ where: { type: 'copy320', sourceFile: uniqSourceFiles[0] }, raw: true })
```

Comment:
ah just realized htis is actually not consistent with the `fileRecord` format from the original check on line 554, which only returns the `multihash` attribute but i guess it doesn't matter